add introduction section to the official CVE feed README (#168)
kubernetes/sig-security: sig-security-tooling/cve-feed/README.md

# Official CVE Feed

## Introduction

The **Kubernetes Official CVE Feed** provides an authoritative and machine-readable source of information about security vulnerabilities (CVEs) affecting Kubernetes.  

> It helps developers, operators, and security professionals stay informed about officially recognized and triaged vulnerabilities in the Kubernetes ecosystem.

Reviewer: I'll put that with the other paragraph before, maybe not a newline?

triaged vulnerabilities in the Kubernetes ecosystem.

This feed is maintained by the **Kubernetes Security Response Committee
(SRC)** and **SIG Security**, ensuring that all CVE data published is
verified and accurately reflects the current security state of
Kubernetes.

### Who Uses the Official CVE Feed

The feed serves multiple audiences:

- **Developers and maintainers** — to identify and track vulnerabilities
  related to their components or dependencies.
- **Security teams and researchers** — to integrate verified Kubernetes
  CVE data into vulnerability management tools or monitoring systems.
- **Cloud providers and Kubernetes distributors** — to automate tracking
  of upstream advisories and coordinate patching.
- **End-users and operators** — to monitor official Kubernetes CVEs and
  apply security updates or mitigations accordingly.

### What It Provides

The Official Kubernetes CVE Feed is distributed in multiple formats:

- **HTML** — a human-readable web page available at  
  [k8s.io/docs/reference/issues-security/official-cve-feed](https://kubernetes.io/docs/reference/issues-security/official-cve-feed/)

> - **JSON** — the original, machine-readable feed listing all issues
>   labeled as
>   [`official-cve-feed`](https://github.com/kubernetes/kubernetes/issues?q=is%3Aissue+label%3Aofficial-cve-feed+)
>   in the
>   [kubernetes/kubernetes](https://github.com/kubernetes/kubernetes)
>   repository
> - **RSS** — a syndication-friendly view for integration into monitoring
>   or alerting tools
> 
> Each entry in these feeds corresponds to an official CVE affecting
> Kubernetes.

Reviewer: ```suggestion
- **JSON** — the original machine-readable feed available at
   [k8s.io/docs/reference/issues-security/official-cve-feed/index.json](https://kubernetes.io/docs/reference/issues-security/official-cve-feed/index.json).
- **RSS** — a syndication-friendly view for integration into monitoring
  or alerting tools available at 
  [k8s.io/docs/reference/issues-security/official-cve-feed/feed.xml](https://kubernetes.io/docs/reference/issues-security/official-cve-feed/feed.xml).

Each entry in these feeds corresponds to an official CVE affecting
Kubernetes retrieved from an issue labeled as 
[`official-cve-feed`](https://github.com/kubernetes/kubernetes/issues?q=is%3Aissue+label%3Aofficial-cve-feed+)
in the [kubernetes/kubernetes](https://github.com/kubernetes/kubernetes)
repository.
```

Author: thank you Mahe, your format is better than me, i will use it

## The official CVE feed is separated into two main components

1. The scripts that update a cloud bucket containing the feed.
2. The website rendering and serving the feed in various formats.

> 
> 

Reviewer: nit: remove double whiteline
```suggestion

```

### Scripts

> A script in the
> [kubernetes/sig-security](https://github.com/kubernetes/sig-security)

Reviewer: it's okay to go over 80 chars when you have links if you want btw, not a stricti rule, the wrapping is just to ease reviewing and git history change by line
```suggestion
A script in the [kubernetes/sig-security](https://github.com/kubernetes/sig-security)
```

repository under the
[sig-security-tooling/cve-feed/hack](https://github.com/kubernetes/sig-security/tree/main/sig-security-tooling/cve-feed/hack)
folder is responsible for generating and updating the feed.

This bash script, named `fetch-cve-feed.sh`:
- sets up the Python 3 environment;
- generates the CVE feed file with `fetch-official-cve-feed.py`;
- compares the sha256 of the newly generated file with the existing one;
- if the sha256 changed, uploads the newly generated CVE feed file to
  the bucket.

The `fetch-official-cve-feed.py` file executed by
`fetch-cve-feed.sh` is a Python 3 script that:
- queries the GitHub API to fetch all issues with the `official-cve-feed`
  label in the
  [kubernetes/kubernetes](https://github.com/kubernetes/kubernetes/issues?q=is%3Aissue%20label%3Aofficial-cve-feed%20)
  repository;
- formats the result according to the JSON Feed schema;
- prints the output to stdout.

These scripts are run regularly as a CronJob on the Kubernetes

> infrastructure.

Reviewer: pplease do not touch unchanged lines (these lines + everything that you wrapped until reference I think) (so that we don't mess up git history)

Author: ok, i did it by mistake, i will keep them untouched next commit

Author: just a quick question @mtardy could i close this pr and open a clean one instead? 
if not, should i revert these lines back to the original or just keep them untouched the next commit?,  i prefer to close this PR and just open a clean one

Reviewer 2: No harm in closing this PR and opening a clean one with all the fixes, if that's what you prefer.

Reviewer: Indeed that's okay if that's the easiest for you.

But otherwise, learning how to rebase a patch set is a nice skill to have: https://docs.github.com/en/get-started/using-git/about-git-rebase. Then you can juste `git push <forkref> HEAD --force-with-lease` and everything should look good. That's how I work personally because on all my main repos we just allow to merge and rebase.

Or you can repush the revert of that part of the commit by `git revert <commitsha>` and then `git reset HEAD~` and then `git add . --patch` and add only what you want to revert. We can then squash everything together at merge time.

or
[storage.googleapis.com/k8s-cve-feed](https://storage.googleapis.com/k8s-cve-feed/).

### Website

The main output of the official CVE feed is the HTML website page
available at
[k8s.io/docs/reference/issues-security/official-cve-feed](https://kubernetes.io/docs/reference/issues-security/official-cve-feed/)
where you can also find links to the JSON and RSS feed formats.

The corresponding HTML page is generated from the
[official-cve-feed.md](https://github.com/kubernetes/website/blob/main/content/en/docs/reference/issues-security/official-cve-feed.md?plain=1)
file in the
[kubernetes/website](https://github.com/kubernetes/website)
repository. It mainly calls the `cve-feed` shortcode defined in
[website/layouts/shortcodes/cve-feed.html](https://github.com/kubernetes/website/blob/main/layouts/shortcodes/cve-feed.html),
which consumes the JSON format by fetching the URL from
[`.Site.Params.cveFeedBucket`](https://github.com/kubernetes/website/blob/75f19fc9675d07fdbc724d02953d905ef7ca8619/hugo.toml#L168)
and rendering it into an HTML table.

This page is updated automatically every time the website is rebuilt.

> 
> 

Reviewer: nit: double whitelines
```suggestion

```
